### PR TITLE
PUB-1624 - Create System admin

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1638

### Change description ###

This includes a new endpoint which isolates the creation of a new system admin account. This was done due to the complexities of logging and current endpoints. The new endpoint bundles the creation of Azure and Pi User into a single endpoint.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by KianKwa — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java [hunk 1]
```diff
@@ -1487,4 +1507,100 @@ void testUpdateAccountByIdNotFound() throws Exception {
...
     }
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:add-system-admin.sql")
+    void testCreateSystemAdminAccount() throws Exception {
+
+        SystemAdminAccount systemAdmin = new SystemAdminAccount();
+        systemAdmin.setFirstName("testSysAdminFirstname");
+        systemAdmin.setSurname("testSysAdminSurname");
+        systemAdmin.setEmail("testSysAdminEmail@gmail.com");
```
Comment (joshblackmoor) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java:1517]: Can we change this email to a fake hmcts.net or justice.gov.uk email pls?

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java [hunk 2]
```diff
@@ -1487,4 +1507,100 @@ void testUpdateAccountByIdNotFound() throws Exception {
...
+        );
+    }
+
+    @Test
+    @WithMockUser(username = UNAUTHORIZED_USERNAME, authorities = {UNAUTHORIZED_ROLE})
+    void testUnauthorizedCreateSystemAdminAccount() throws Exception {
+        SystemAdminAccount systemAdmin = new SystemAdminAccount();
+        systemAdmin.setFirstName("testSysAdminFirstname");
+        systemAdmin.setSurname("testSysAdminSurname");
+        systemAdmin.setEmail("testSysAdminEmail@gmail.com");
```
Comment (joshblackmoor) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java:1548]: Same comment as above on email

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java [hunk 3]
```diff
@@ -1487,4 +1507,100 @@ void testUpdateAccountByIdNotFound() throws Exception {
...
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:add-system-admin.sql")
+    void testCreateSystemAdminAccountRequestExceeded() throws Exception {
+
+        SystemAdminAccount systemAdmin1 = new SystemAdminAccount();
+        systemAdmin1.setFirstName("testSysAdminFirstname1");
+        systemAdmin1.setSurname("testSysAdminSurname1");
+        systemAdmin1.setEmail("testSysAdminEmai1l@gmail.com");
```
Comment (joshblackmoor) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java:1572]: Same comment as above on email

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java [hunk 4]
```diff
@@ -1487,4 +1507,100 @@ void testUpdateAccountByIdNotFound() throws Exception {
...
+                .header(ISSUER_HEADER, SYSTEM_ADMIN_ISSUER_ID)
+                .contentType(MediaType.APPLICATION_JSON);
+
+        mockMvc.perform(createRequest1)
+            .andExpect(status().isOk());
+
+        SystemAdminAccount systemAdmin2 = new SystemAdminAccount();
+        systemAdmin2.setFirstName("testSysAdminFirstname2");
+        systemAdmin2.setSurname("testSysAdminSurname2");
+        systemAdmin2.setEmail("testSysAdminEmai12@gmail.com");
```
Comment (joshblackmoor) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java:1589]: Same comment as above on email

File: src/functionalTest/resources/add-system-admin.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO pi_user (user_id, email, provenance_user_id,user_provenance,roles,forenames,surname)
+VALUES ('87f907d2-eb28-42cc-b6e1-ae2b03f7bba2', 'SyestemAdmin@gmail.com', 'e5f1cc77-6e9a-40ab-8da0-a9666b328464','PI_AAD','SYSTEM_ADMIN','System','Admin');
```
Comment (joshblackmoor) [inline src/functionalTest/resources/add-system-admin.sql:2]: Could the email in here also be changed pls

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java (hunk 3 above)
Comment (ashwini-mv) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java]: Done

File: src/functionalTest/resources/add-system-admin.sql (hunk 5 above)
Comment (ashwini-mv) [reply src/functionalTest/resources/add-system-admin.sql]: Done

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java (hunk 4 above)
Comment (ashwini-mv) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java]: Done

**Review by joshblackmoor — APPROVED** [review]

**Pull request merged**